app/models/post.rb

```diff
@@ -0,0 +1,27 @@
+class Post < ApplicationRecord
+  # Establishes a belongs_to association with the User model, using author_id as the foreign key.
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+
+  # Establishes has_many associations with the Comment and Like models.
+  has_many :comments
+  has_many :likes
+
+  # Validations for the Post model
+  validates :title, presence: true
+  validates :text, presence: true
```

Comment:
- [ ] As mentioned by all previous reviewers you need to add validations for title to be not more than 250 characters and counters to be of a valid type, i.e. counters to only integers. For example, 
 
```rb

  validates :title, presence: true, length: { in: 1..250 }
  validates :comments_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
  validates :likes_counter, numericality: { only_integer: true, greater_than_or_equal_to: 0 }
```